plugins/woocommerce/includes/abstracts/abstract-wc-order.php

```diff
@@ -1313,6 +1313,15 @@ public function remove_coupon( $code ) {
 				$coupon_object->decrease_usage_count( $this->get_user_id() );
 				$this->recalculate_coupons();
 
+				/**
+				 * Action hook to indicate that a coupon has been removed from an order.
+				 *
+				 * @param WC_Coupon $coupon The coupon that was removed.
+				 * @param WC_Order  $order  The order that the coupon was removed from.
+				 */
+				do_action( 'woocommerce_order_removed_coupon', $coupon_object, $this );
```

Comment:
```suggestion
				 * @param string             $coupon_code The coupon that was removed.
				 * @param WC_Abstract_Order  $order  The order that the coupon was removed from.
				 */
				do_action( 'woocommerce_order_removed_coupon', $code, $this );
```

If we want to have [parity](https://github.com/woocommerce/woocommerce/issues/36114) with the `woocommerce_removed_coupon` action in the cart (so that you could, for example, run the same callback on both hooks), the first parameter should probably be the `$code` string instead of the full object.